dagshub/data_engine/model/datasource.py

```diff
@@ -543,6 +551,13 @@ def _upload_metadata(self, metadata_entries: List[DatapointMetadataUpdateEntry])
         # Update the status from dagshub, so we get back the new metadata columns
         self.source.get_from_dagshub()
 
+    def delete_metadata(self, path, id, key):
+        metadata_entries = [DatapointDeleteMetadataEntry(datapointId=id, url=path, key=key)]
+        self.source.client.delete_metadata(self, metadata_entries)
+
+        # Update the status from dagshub, so we get back the new metadata columns
+        self.source.get_from_dagshub()
+
```

Comment:
Considering that we also want to be able to delete the whole field from the datasource at once, this needs to be named differently at the very least.

Delete field from the whole datasource: `delete_metadata`
Delete field from just one datapoint: `delete_metadata_from_datapoint`

But honestly I think it's fine if this function exists just on the datapoint object, even if that kind of forces users to do a query before deletion.

Another problem with this function: `id` is not user-usable, because it requires a user to go through a query anyway. If you _do_ want to keep this function, then either the GQL mutation shouldn't accept BOTH path and id, but either of them instead, 
Alternative that doesn't require redesigning the GQL mutation [less preferrable IMO]: do a roundtrip here with `(ds[path] == "<url>").head()[0]`. (I don't think we still have this as a "neat" function like `get_datapoint()` that you could use). Also keep in mind that `self` also carries with it a filter, so doing a query like that would require a copy of the `self` with its filter reset
Alternative alternative: make this function accept `Datapoint` instead of `ID+Path`

Reply:
i ment this to not be a user function , i should rename it _delete_metadata.
the api is datapoint.delete_metadata(name)